src/libbson/src/bson/bson-string.c

```diff
@@ -116,7 +116,46 @@ bson_string_new (const char *str) /* IN */
 
    ret->str[len_u32] = '\0';
    ret->len = len_u32;
+   return ret;
+}
 
+/*
+ *--------------------------------------------------------------------------
+ *
+ * _bson_string_alloc --
+ *
+ *       Create an empty bson_string_t and allocate memory for it.
+ *
+ *       The amount of memory allocated will be the next power-of-two if the
+ *       specified size is not already a power-of-two.
+ *
+ * Parameters:
+ *       @size: Size of the string to allocate
+ *
+ * Returns:
+ *       A newly allocated bson_string_t that should be freed with
+ *       bson_string_free().
+ *
+ * Side effects:
+ *       None.
+ *
+ *--------------------------------------------------------------------------
+ */
+
+bson_string_t *
+_bson_string_alloc (const size_t size)
+{
+   BSON_ASSERT (bson_in_range_unsigned (uint32_t, size));
```

Comment:
```suggestion
   BSON_ASSERT (size < UINT32_MAX);
```

`bson_string_ensure_space` asserts `BSON_ASSERT (needed <= UINT32_MAX - 1u);`. Reuse that assertion here to leave room for guaranteed null terminator.